sql/analyzer/exec_builder.go

```diff
@@ -110,6 +110,15 @@ func (b *ExecBuilder) buildLookup(l *lookup, input sql.Schema, children ...sql.N
 			ret = plan.NewTableAlias(n.Name(), ret)
 		}
 		ret = plan.NewDistinct(ret)
+	case *plan.Filter:
```

Comment:
Organizational simplification opp: if you made filter, distinct, and SelectSingleRel implement `sql.UnaryNode`, and add `UnaryNode.WithChild()`, all of these could condense to this:
```golang
	case sql.UnaryNode:
		switch n := n.Child().(type) {
		case *plan.ResolvedTable:
			ret, err = plan.NewIndexedAccessForResolvedTable(n, plan.NewLookupBuilder(l.index, keyExprs, l.nullmask))
		case *plan.TableAlias:
			ret, err = plan.NewIndexedAccessForResolvedTable(n.Child.(*plan.ResolvedTable), plan.NewLookupBuilder(l.index, keyExprs, l.nullmask))
			ret = plan.NewTableAlias(n.Name(), ret)
		}
		ret = n.WithChild(ret)
```

Reply:
`*plan.SelectSingleRel` is also kind of defunct now, could just attach the filter to the exprGroup's select. Which would also let us delete the `normalizeSelectSingleRel` rule.

Reply:
Neither of these are blocking for this PR

Reply:
I tried changing filter and distinct to implement `sql.UnaryNode`, but this requires a lot of changes to be made due to go's composition: `filter.Child` used to return `filter.UnaryNode.Child`, but now `Child` is an un-invoked function.
So every place we ever used `filter.Child`/`distinct.Child`, now have to be updated to `filter.Child()`/`distinct.Child()`. These are changes we'd have to make to both GMS and dolt.

But the worst thing is that we don't get notified of all of these cases at build time: [this particular line](https://github.com/dolthub/go-mysql-server/blob/main/sql/plan/filter.go#L92) does not generate a build-time error, since `DebugString` accepts *anything*, but this line will cause a run-time error.

So changing filter and distinct to implement `sql.UnaryNode` is quite risky. For now, I'm going to leave a todo in exec_builder with this suggestion.

I haven't tried the `*plan.SelectSingleRel`, but that one also sounds pretty involved. But I can add a todo to `normalizeSelectSingleRel` for us to consider removing it.